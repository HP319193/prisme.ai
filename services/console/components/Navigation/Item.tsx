import Link from 'next/link';
import { useRouter } from 'next/router';
import { FC, ReactChild, ReactElement } from 'react';

export interface ItemProps {
  href: string;
  icon: ReactChild | ((props: { selected: Boolean }) => ReactElement);
}
export const Item: FC<ItemProps> = ({ href, icon: Icon, children }) => {
  const { asPath } = useRouter();
  const selected = decodeURIComponent(asPath) === href;
  return (
    <Link href={href}>
      <a
        className={`flex flex-1 leading-10 px-4 py-2 group hover:bg-ultra-light-accent !text-base ${
          selected ? 'bg-ultra-light-accent font-bold' : ''
        }`}
      >
        <div
          className={`flex flex-1 flex-row items-center ${
            selected ? 'text-accent' : ''
          } max-w-[85%]`}
        >
          <div className="flex m-2 mr-4">
            {typeof Icon === 'function' ? <Icon selected={selected} /> : Icon}
          </div>
          <div className="flex flex-1 leading-7 max-w-full">{children}</div>
        </div>
      </a>
    </Link>
  );
};

export default Item;
