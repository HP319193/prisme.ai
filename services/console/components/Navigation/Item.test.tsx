import renderer from 'react-test-renderer';
import Item from './Item';

jest.mock(
  'next/link',
  () =>
    function Link(props: any) {
      return <div id="next/link" {...props} />;
    }
);

jest.mock('next/router', () => {
  const mock = {
    asPath: '/foo/bar',
  };
  return {
    useRouter: () => mock,
  };
});

it('should render', () => {
  const root = renderer.create(
    <Item href="/foo" icon={<div>Icon</div>}></Item>
  );
  expect(root.toJSON()).toMatchSnapshot();
});

it('should render selected', () => {
  const root = renderer.create(
    <Item href="/foo/bar" icon={<div>Icon</div>}></Item>
  );
  expect(root.toJSON()).toMatchSnapshot();
});
