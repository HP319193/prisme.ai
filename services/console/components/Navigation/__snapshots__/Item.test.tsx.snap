// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div
  href="/foo"
  id="next/link"
>
  <a
    className="flex flex-1 leading-10 px-4 py-2 group hover:bg-ultra-light-accent !text-base whitespace-nowrap "
  >
    <div
      className="flex flex-1 flex-row items-center  max-w-[85%]"
    >
      <div
        className="flex m-2 mr-4"
      >
        <div>
          Icon
        </div>
      </div>
      <div
        className="flex flex-1 leading-7 max-w-full"
      />
    </div>
  </a>
</div>
`;

exports[`should render selected 1`] = `
<div
  href="/foo/bar"
  id="next/link"
>
  <a
    className="flex flex-1 leading-10 px-4 py-2 group hover:bg-ultra-light-accent !text-base whitespace-nowrap bg-ultra-light-accent font-bold"
  >
    <div
      className="flex flex-1 flex-row items-center text-accent max-w-[85%]"
    >
      <div
        className="flex m-2 mr-4"
      >
        <div>
          Icon
        </div>
      </div>
      <div
        className="flex flex-1 leading-7 max-w-full"
      />
    </div>
  </a>
</div>
`;
